src/vm/prestub.cpp

```diff
@@ -1001,7 +1001,7 @@ BOOL PrepareCodeConfig::MayUsePrecompiledCode()
 VersionedPrepareCodeConfig::VersionedPrepareCodeConfig() {}
 
 VersionedPrepareCodeConfig::VersionedPrepareCodeConfig(NativeCodeVersion codeVersion) :
-    PrepareCodeConfig(codeVersion, TRUE, FALSE)
+    PrepareCodeConfig(codeVersion, TRUE, codeVersion.GetOptimizationTier() == NativeCodeVersion::OptimizationTier0)
```

Comment:
It was the case (or it was supposed to be the case) that VersionedPrepareCodeConfig was only created for non-default code versions (line 262). If that invariant still holds then this new check should be unnecessary. If something got rid of that invariant then the check we'd want to use here is codeVersion.IsDefaultCodeVersion().

It is possible to have the profiler do IL rewriting which produces tier0 non-default code. That code version should not be using precompiled code.

Reply:
Oh ok, I had this in my in-progress changes for a while, let me see again why I thought it was needed

Reply:
I must have misread it, I don't see it going down this path